src/utils/date.js

```diff
@@ -57,9 +60,9 @@ export function getDate(value) {
 /** Map from attribute names to their parsers. */
 const dateAttrParsers = {
   'datetime': (datetime) =>
-    pureUserAssert(parseDate(datetime), `Invalid date: ${datetime}`),
+    userAssert(parseDate(datetime), 'Invalid date: %s', datetime),
```

Comment:
FMI: Why is this format preferred to template literals?

Reply:
There is partial work (according to @erwinmombay it's mostly complete) around having Babel remove all assertion strings, and this would end up generating a URL to an error page, something like `https://cdn.ampproject.org/error/1234?v1=${datetime}`. Error strings themselves get compiled out and replaced with error codes, saving some bytes along the way. I don't know where the I2I or docs around this are at the moment.

Reply:
That sort of makes sense. But another reason why this is useful: we avoid constructing the error string if `userAssert` does not throw an error. And we expect that throwing an error here would be an exception (pun semi-intended).

Reply:
I believe we supported both forms, you just couldn't mix `%s` inside a template literal.